src/ray/core_worker/task_interface.cc

```diff
@@ -1,26 +1,167 @@
-  return Status::OK();
+  auto &context = core_worker_.worker_context_;
+  auto next_task_index = context.GetNextTaskIndex();
+  const auto task_id = GenerateTaskId(context.GetCurrentDriverID(),
+                                      context.GetCurrentTaskID(), next_task_index);
+
+  // add one for actor cursor object id.
+  auto num_returns = task_options.num_returns + 1;
+  (*return_ids).resize(num_returns);
+  for (int i = 0; i < num_returns; i++) {
+    (*return_ids)[i] = ObjectID::ForTaskReturn(task_id, i + 1);
+  }
+
+  auto actor_creation_dummy_object_id =
+      ObjectID::FromBinary(actor_handle.ActorID().Binary());
+
+  auto task_arguments = BuildTaskArguments(args);
+  ::Language language;
+  auto status = ToTaskLanguage(function.language, &language);
+  if (!status.ok()) {
+    RAY_LOG(ERROR) << status.message() << "  task: " << task_id
+                   << ", language: " << static_cast<int>(function.language);
+    return status;
+  }
+
+  std::vector<ActorHandleID> new_actor_handles;
+  ray::raylet::TaskSpecification spec(
+      context.GetCurrentDriverID(), context.GetCurrentTaskID(), next_task_index,
+      ActorID::Nil(), actor_creation_dummy_object_id, 0, actor_handle.ActorID(),
+      actor_handle.ActorHandleID(), actor_handle.IncreaseTaskCounter(), new_actor_handles,
+      task_arguments, num_returns, task_options.resources, task_options.resources,
+      language, function.function_descriptor);
+
+  std::vector<ObjectID> execution_dependencies;
+  execution_dependencies.push_back(actor_handle.ActorCursor());
+
+  auto actor_cursor = (*return_ids).back();
+  actor_handle.SetActorCursor(actor_cursor);
+  actor_handle.ClearNewActorHandles();
+
+  status = core_worker_.raylet_client_->SubmitTask(execution_dependencies, spec);
+
+  // remove cursor from return ids.
+  (*return_ids).pop_back();
+  return status;
+}
+
+std::vector<std::shared_ptr<raylet::TaskArgument>>
+CoreWorkerTaskInterface::BuildTaskArguments(const std::vector<TaskArg> &args) {
+  std::vector<std::shared_ptr<raylet::TaskArgument>> task_arguments;
+  for (const auto &arg : args) {
+    if (arg.IsPassedByReference()) {
+      std::vector<ObjectID> references{arg.GetReference()};
+      task_arguments.push_back(
+          std::make_shared<raylet::TaskArgumentByReference>(references));
+    } else {
+      auto data = arg.GetValue();
+      task_arguments.push_back(
+          std::make_shared<raylet::TaskArgumentByValue>(data->Data(), data->Size()));
+    }
+  }
+  return task_arguments;
+}
+
+Status CoreWorkerTaskInterface::ToTaskLanguage(
+    WorkerLanguage language, ::Language *out) {
+  switch (language) {
+  case ray::WorkerLanguage::JAVA:
+    *out = ::Language::JAVA;
+    return Status::OK();
+    break;
+  case ray::WorkerLanguage::PYTHON:
+    *out = ::Language::PYTHON;
+    return Status::OK();
+    break;
+  default:
+    *out = ::Language::PYTHON;
+    return Status::Invalid("invalid language specified");
+    break;
```

Comment:
I think it's okay to just crash here if the language is invalid. Because this could happen only when there's a bug in c++ core worker. 
I think the principle should be the following. For errors that are only due to code bugs, we should just use `RAY_CHECK` or `RAY_LOG(FATAL)` to crash. For other errors (i.g., network error), we should return status and propagate error to Java.
Does this make sense?

Reply:
It's slightly different that in this case this could also happen if there's a bug in java/python worker, in that theoretically they can cast a `int` into `WorkerLanguage`, and then call the `TaskInterface` in core worker. That said, I agree in this case we can just crash as this is indeed a bug in the code, not uncontrollable errors.

Reply:
Also it would make the code simpler. will change as suggested.